### PR TITLE
Minor changes in Azure Experiments

### PR DESCRIPTION
**Pull request by avaakash**
**What this PR does / why we need it**:
This PR changes the function parameters for the Instance Check method in the Azure instance stop experiment and Disk Check method in Azure disk loss experiment for usage in other azure instance-related experiments.
It also removes the AUT and Auxillary App check from the Azure experiments as they are not required.
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
-   [ ] Fixes #<issue number>
-   [ ] PR messages has document related information
-   [ ] Labelled this PR & related issue with `breaking-changes` tag
-   [ ] PR messages has breaking changes related information
-   [ ] Labelled this PR & related issue with `requires-upgrade` tag
-   [ ] PR messages has upgrade related information
-   [ ] Commit has unit tests
-   [ ] Commit has integration tests
-   [ ] E2E run Required for the changes

### CODE REVIEW

**Comment by avaakash** [comment]
## Experiment Logs - Azure Instance Stop
```
INFO[2022-01-05T16:19:37+05:30] Experiment Name: azure-instance-stop         
INFO[2022-01-05T16:19:37+05:30] [PreReq]: Getting the ENV for the azure-instance-stop experiment 
INFO[2022-01-05T16:19:37+05:30] [PreReq]: Updating the chaos result of azure-instance-stop experiment (SOT) 
INFO[2022-01-05T16:19:39+05:30] The application information is as follows     Instance Name=akash-run-command Chaos Duration=30 Resource Group=akash-litmus-test
INFO[2022-01-05T16:19:39+05:30] [Info]: The instance under chaos(IUC) are: [akash-run-command] 
INFO[2022-01-05T16:19:40+05:30] [Status]: The instance akash-run-command state is: 'VM running' 
INFO[2022-01-05T16:19:40+05:30] [Status]: Azure instance(s) is in running state (pre-chaos) 
INFO[2022-01-05T16:19:40+05:30] [Info]: Target instanceName list, [akash-run-command] 
INFO[2022-01-05T16:19:40+05:30] [Chaos]: Stopping the Azure instance: akash-run-command 
INFO[2022-01-05T16:19:40+05:30] [Info]: Stopping the instance                
INFO[2022-01-05T16:19:41+05:30] [Wait]: Waiting for Azure instance 'akash-run-command' to get in the stopped state 
INFO[2022-01-05T16:19:41+05:30] [Status]: Checking Azure instance status     
INFO[2022-01-05T16:19:42+05:30] [Status]: The instance akash-run-command state is: 'VM stopping' 
INFO[2022-01-05T16:19:44+05:30] [Status]: The instance akash-run-command state is: 'VM stopping' 
INFO[2022-01-05T16:19:47+05:30] [Status]: The instance akash-run-command state is: 'VM stopped' 
INFO[2022-01-05T16:19:49+05:30] [Wait]: Waiting for chaos interval of 30s    
INFO[2022-01-05T16:20:19+05:30] [Chaos]: Starting back the Azure instance: akash-run-command 
INFO[2022-01-05T16:20:19+05:30] [Info]: Starting back the instance to running state 
INFO[2022-01-05T16:20:21+05:30] [Wait]: Waiting for Azure instance 'akash-run-command' to get in the running state 
INFO[2022-01-05T16:20:21+05:30] [Status]: Checking Azure instance status     
INFO[2022-01-05T16:20:22+05:30] [Status]: The instance akash-run-command state is: 'VM starting' 
INFO[2022-01-05T16:20:25+05:30] [Status]: The instance akash-run-command state is: 'VM running' 
INFO[2022-01-05T16:20:27+05:30] [Confirmation]: Azure instance stop chaos has been injected successfully 
INFO[2022-01-05T16:20:27+05:30] [Info]: The instance under chaos(IUC) are: [akash-run-command] 
INFO[2022-01-05T16:20:28+05:30] [Status]: The instance akash-run-command state is: 'VM running' 
INFO[2022-01-05T16:20:28+05:30] [Status]: Azure instance is in running state (post chaos) 
INFO[2022-01-05T16:20:28+05:30] [The End]: Updating the chaos result of azure-instance-stop experiment (EOT) 
```

**Comment by avaakash** [comment]
## Experiment Logs - Azure Disk Loss
```
INFO[2022-01-05T16:22:57+05:30] Experiment Name: azure-disk-loss             
INFO[2022-01-05T16:22:57+05:30] [PreReq]: Getting the ENV for the azure-disk-loss experiment 
INFO[2022-01-05T16:22:57+05:30] [PreReq]: Updating the chaos result of azure-disk-loss experiment (SOT) 
INFO[2022-01-05T16:23:02+05:30] The volume information is as follows          Sequence=parallel Disk Names="test-disk-1,test-disk-2" Resource Group=akash-litmus-test
INFO[2022-01-05T16:23:02+05:30] [Status]: Verify that the virtual disk are attached to VM instance(pre-chaos) 
INFO[2022-01-05T16:23:05+05:30] [Chaos]: Detaching the virtual disks from the instances 
INFO[2022-01-05T16:23:12+05:30] [Wait]: Waiting for Disk 'test-disk-1' to detach 
INFO[2022-01-05T16:23:13+05:30] [Status]: Disk test-disk-1 is not yet detached, state: Attached 
INFO[2022-01-05T16:23:16+05:30] [Status]: Disk test-disk-1 is not yet detached, state: Attached 
INFO[2022-01-05T16:23:18+05:30] [Status]: Disk test-disk-1 is not yet detached, state: Attached 
INFO[2022-01-05T16:23:21+05:30] [Status]: Disk test-disk-1 is Detached       
INFO[2022-01-05T16:23:23+05:30] [Wait]: Waiting for Disk 'test-disk-2' to detach 
INFO[2022-01-05T16:23:24+05:30] [Status]: Disk test-disk-2 is Detached       
INFO[2022-01-05T16:23:26+05:30] [Wait]: Waiting for the chaos interval of 30s 
INFO[2022-01-05T16:23:56+05:30] [Chaos]: Attaching the Virtual disks back to the instances 
INFO[2022-01-05T16:24:00+05:30] [Wait]: Waiting for Disk 'test-disk-1' to attach 
INFO[2022-01-05T16:24:01+05:30] [Status]: Disk test-disk-1 is Attached       
INFO[2022-01-05T16:24:03+05:30] [Wait]: Waiting for Disk 'test-disk-2' to attach 
INFO[2022-01-05T16:24:04+05:30] [Status]: Disk test-disk-2 is Attached       
INFO[2022-01-05T16:24:06+05:30] [Confirmation]: azure-disk-loss chaos has been injected successfully 
INFO[2022-01-05T16:24:06+05:30] [Status]: Verify that the virtual disk are attached to VM instance(post-chaos) 
INFO[2022-01-05T16:24:07+05:30] [The End]: Updating the chaos result of azure-disk-loss experiment (EOT) 
```

**Review by neelanjan00 — DISMISSED** [review]

File: pkg/azure/disk-loss/environment/environment.go [hunk 1]
```diff
@@ -18,19 +18,13 @@ func GetENV(experimentDetails *experimentTypes.ExperimentDetails) {
...
 	experimentDetails.ChaosUID = clientTypes.UID(types.Getenv("CHAOS_UID", ""))
 	experimentDetails.InstanceID = types.Getenv("INSTANCE_ID", "")
 	experimentDetails.ChaosPodName = types.Getenv("POD_NAME", "")
 	experimentDetails.Delay, _ = strconv.Atoi(types.Getenv("STATUS_CHECK_DELAY", "2"))
 	experimentDetails.Timeout, _ = strconv.Atoi(types.Getenv("STATUS_CHECK_TIMEOUT", "180"))
-	experimentDetails.TargetContainer = types.Getenv("TARGET_CONTAINER", "")
 	experimentDetails.ScaleSet = types.Getenv("SCALE_SET", "disable")
 	experimentDetails.ResourceGroup = types.Getenv("RESOURCE_GROUP", "")
 	experimentDetails.VirtualDiskNames = types.Getenv("VIRTUAL_DISK_NAMES", "")
-	experimentDetails.SubscriptionID = types.Getenv("SUBSCRIPTION_ID", "")
```
Comment (oumkale) [inline pkg/azure/disk-loss/environment/environment.go:34]: PTAL, is used in experiments

Comment (avaakash) [reply pkg/azure/disk-loss/environment/environment.go:34]: We are setting up the Subscription ID later, not through environment.

File: experiments/azure/azure-disk-loss/experiment/azure-disk-loss.go [hunk 2]
```diff
@@ -74,26 +73,6 @@ func AzureDiskLoss(clients clients.ClientSets) {
...
-	if experimentsDetails.AuxiliaryAppInfo != "" {
-		log.Info("[Status]: Verify that the Auxiliary Applications are running (pre-chaos)")
-		if err = status.CheckAuxiliaryApplicationStatus(experimentsDetails.AuxiliaryAppInfo, experimentsDetails.Timeout, experimentsDetails.Delay, clients); err != nil {
-			log.Errorf("Auxiliary Application status check failed, err: %v", err)
-			failStep := "[pre-chaos]: Failed to verify that the Auxiliary Applications are in running state, err: " + err.Error()
-			result.RecordAfterFailure(&chaosDetails, &resultDetails, failStep, clients, &eventsDetails)
-			return
-		}
-	}
-
```
Comment (uditgaurav) [inline experiments/azure/azure-disk-loss/experiment/azure-disk-loss.go:96]: Why are we removing the support for app and aux-app check for azure experiments? The experiment (disk-loss, instance stop) can also impact the state of the application.

File: experiments/azure/azure-disk-loss/experiment/azure-disk-loss.go [hunk 3]
```diff
@@ -104,7 +83,7 @@ func AzureDiskLoss(clients clients.ClientSets) {
 
 	// PRE-CHAOS VIRTUAL DISK STATUS CHECK
 	log.Info("[Status]: Verify that the virtual disk are attached to VM instance(pre-chaos)")
-	if err = azureStatus.CheckVirtualDiskWithInstance(experimentsDetails); err != nil {
+	if err = azureStatus.CheckVirtualDiskWithInstance(experimentsDetails.SubscriptionID, experimentsDetails.VirtualDiskNames, experimentsDetails.ResourceGroup); err != nil {
```
Comment (uditgaurav) [inline experiments/azure/azure-disk-loss/experiment/azure-disk-loss.go:86]: line 93: Could be just after the AUT check function

File: experiments/azure/azure-disk-loss/experiment/azure-disk-loss.go (hunk 2 above)
Comment (avaakash) [reply experiments/azure/azure-disk-loss/experiment/azure-disk-loss.go]: Re-added the AUT and aux-app check.

**Review by uditgaurav — APPROVED** [review]

**Pull request merged**